### PR TITLE
Move SDK reporting to separate workflow job

### PR DESCRIPTION
**Pull request by shamilovtim**
### Context
Ever since the Junit reporter was introduced the `test-with-node` CI workflow was silencing human readable test results which was causing pain for devs.

## Solution
Separate workflow for reporting.

I could have either introduced mocha multi reporters (an old package with bugs) or a separate workflow. I went with the separate workflow because it reduces complexity, doesn't introduce new dependencies (especially old ones) and actually separates concerns nicely. Furthermore, GH linux containers are free and plentiful so wasting resources here is not a concern.

By having a separate workflow the concerns are separated:

1. The SDK reporting run focuses on SDK reporting only with an XML reporter (not for use by devs)
2. The human readable reporter run focuses on linting, testing, and coverage (for use by devs).

### CODE REVIEW

**Comment by frankhinek** [comment]
@shamilovtim Renamed the PR title to describe the intent of the PR

@shamilovtim and @finn-tbd @leordev In thinking about this workflow, does it make sense to have the SDK reporting live as a complete separate workflow that runs _solely_ on merges to main?

In other words, should we be updating the spec compliance report that lives [here](https://tbd54566975.github.io/sdk-report-runner/) based on unapproved PRs?

**Comment by shamilovtim** [comment]
> @shamilovtim Renamed the PR title to describe the intent of the PR
> 
> @shamilovtim and @finn-tbd @leordev In thinking about this workflow: What are your thoughts on moving SDK spec compliance [reporting](https://tbd54566975.github.io/sdk-report-runner/) to a separate workflow that runs _solely_ on merges to main? In other words, should we be updating the spec compliance report based on unapproved PRs?

I have made the change for this repo and @finn-tbd and @leordev can coordinate the other ones if we're all in agreement

**Review by nitro-neal — APPROVED** [review]
Thank you for doing this, makes sense to move it to separate file and only on main as far as report uploading goes

**Comment by shamilovtim** [comment]
piggybacked splitting the web tests on this PR

**Comment by finn-tbd** [comment]
> What are your thoughts on moving SDK spec compliance [reporting](https://tbd54566975.github.io/sdk-report-runner/) to a separate workflow that runs solely on merges to main? In other words, should we be updating the spec compliance report based on unapproved PRs?

fine by me as long as the validation that it's not going to break is still being run before merges to main. Presumably we're just talking about the parsing step here, which is fine to run after merge.

**Review by finn-tbd — APPROVED** [review]

**Review by leordev — APPROVED** [review]

File: .github/workflows/reports.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  node-sdk-reports:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+        with:
+          submodules: true
+
+      - name: Init Hermit
+        uses: cashapp/activate-hermit@v1
```
Comment (frankhinek) [inline .github/workflows/reports.yml:21]:
```suggestion
        uses: cashapp/activate-hermit@31ce88b17a84941bb1b782f1b7b317856addf286 #v1.1.0
```

File: .github/workflows/tests-ci.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
         with:
           submodules: true
-    
+
       # https://cashapp.github.io/hermit/usage/ci/
       - name: Init Hermit
         uses: cashapp/activate-hermit@v1
```
Comment (frankhinek) [inline .github/workflows/tests-ci.yml:40]:
```suggestion
        uses: cashapp/activate-hermit@31ce88b17a84941bb1b782f1b7b317856addf286 #v1.1.0
```

**Pull request merged**